nautobot_ssot/contrib/sorting.py

```diff
@@ -0,0 +1,105 @@
+"""Functions for sorting DiffSync model lists ensuring they are sorted to prevent false actions."""
```

Comment:
I think a lot of this and of the existing base code should at some point be refactored to use [annotationlib](https://docs.python.org/3.14/library/annotationlib.html) - this is not for now though, maybe you can just leave a comment?